setupext.py

```diff
@@ -375,65 +273,49 @@ def set_pkgconfig_path(self):
         except KeyError:
             os.environ['PKG_CONFIG_PATH'] = pkgconfig_path
 
-    def setup_extension(self, ext, package, default_include_dirs=[],
-                        default_library_dirs=[], default_libraries=[],
-                        alt_exec=None):
-        """
-        Add parameters to the given `ext` for the given `package`.
-        """
-        flag_map = {
-            '-I': 'include_dirs', '-L': 'library_dirs', '-l': 'libraries'}
-
-        executable = alt_exec
-        if self.has_pkgconfig:
-            executable = (self.pkg_config + ' {0}').format(package)
-
-        use_defaults = True
+    def setup_extension(
+            self, ext, package,
+            atleast_version=None, alt_exec=None, default_libraries=()):
+        """Add parameters to the given *ext* for the given *package*."""
 
-        if executable is not None:
-            command = "{0} --libs --cflags ".format(executable)
+        # First, try to get the flags from pkg-config.
 
+        cmd = ([self.pkg_config, package] if self.pkg_config else alt_exec)
+        if cmd is not None:
             try:
-                output = subprocess.check_output(
-                    command, shell=True, stderr=subprocess.STDOUT)
-            except subprocess.CalledProcessError:
+                if self.pkg_config and atleast_version:
+                    subprocess.check_call(
+                        [*cmd, f"--atleast-version={atleast_version}"])
+                # Use sys.getfilesystemencoding() to allow round-tripping
+                # when passed back to later subprocess calls; do not use
+                # locale.getpreferredencoding() which universal_newlines=True
+                # would do.
+                cflags = shlex.split(
+                    os.fsdecode(subprocess.check_output([*cmd, "--cflags"])))
+                libs = shlex.split(
+                    os.fsdecode(subprocess.check_output([*cmd, "--libs"])))
```

Comment:
Is the previously added (but now removed again) comment about `--ldflags` no longer true?

Reply:
For the libpng-config case (so only if pkg-config is not present), that means that the cflags variable will actually also contain the output given by libpng-config --ldflags... which is not a problem (per the now removed comment).